### PR TITLE
Use locations on holdings records to inform the availabilities

### PR DESCRIPTION
**Pull request by alexwlchan**
Closes https://github.com/wellcomecollection/platform/issues/5104

### CODE REVIEW

File: common/internal_model/src/main/scala/weco/catalogue/internal_model/work/Availability.scala [hunk 1]
```diff
@@ -40,28 +40,32 @@ object Availability extends Enum[Availability] {
...
 object Availabilities {
-  def forWorkData(data: WorkData[_]): Set[Availability] =
+  def forWorkData(data: WorkData[_]): Set[Availability] = {
+    val locations =
+      data.items.flatMap { _.locations} ++
+        data.holdings.flatMap { _.location }
+
     Set(
-      when(containsLocation(_.isInstanceOf[PhysicalLocation])(data.items))(
+      when(locations.exists(_.isPhysicalLocation)) {
```
Comment (jamieparkinson) [inline common/internal_model/src/main/scala/weco/catalogue/internal_model/work/Availability.scala:49]: nicerer

File: common/internal_model/src/test/scala/weco/catalogue/internal_model/work/AvailabilityTest.scala [hunk 2]
```diff
@@ -28,13 +28,47 @@ class AvailabilityTest
...
 
-    it("adds Availability.InLibrary if there is a physical location") {
+    it("adds Availability.InLibrary if there is an item with a physical location") {
       val work = denormalisedWork().items(List(createIdentifiedPhysicalItem))
       val workAvailabilities = Availabilities.forWorkData(work.data)
 
       workAvailabilities should contain only Availability.InLibrary
     }
 
+    describe("if there is a holdings") {
```
Comment (jamieparkinson) [inline common/internal_model/src/test/scala/weco/catalogue/internal_model/work/AvailabilityTest.scala:38]: This is grammatically stressful

**Review by jamieparkinson — APPROVED** [review]
🎉

File: common/internal_model/src/test/scala/weco/catalogue/internal_model/work/AvailabilityTest.scala (hunk 2 above)
Comment (alexwlchan) [reply common/internal_model/src/test/scala/weco/catalogue/internal_model/work/AvailabilityTest.scala:38]: evergreen description of the holdings work

**Pull request merged**